### PR TITLE
PAPP-23589 - Add md5 support for generating SDI for non-fips Phantom …

### PR DESCRIPTION
**Pull request by chunmanjimmyf**
…instances

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [X] Testing of all the changes has been performed (for bug fixes / features)
- [X] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [X] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [X] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [X] Verify all checks are passing.
- [X] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [X] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Currently we generate SDIs in SHA256 format when creating artifacts on ingestion.

## What is the new behavior? (OPTIONAL)
- For non-fips Phantom instances, we will generate SDIs in MD5 format so we do not create duplicate artifacts if the artifact already existed prior.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by chunmanjimmyf** [comment]
Integration Test Failure: trace_email_valid is new & requires https://github.com/splunk-soar-connectors/office365/pull/3 to be merged in. All other tests are passing.

File: ewsonprem_connector.py [hunk 1]
```diff
@@ -2590,8 +2590,20 @@ def _process_email_ids(self, email_ids, action_result):
...
 
-    def _poll_now(self, param):
+    def _get_fips_enabled(self):
+        try:
+            from phantom_common.install_info import is_fips_enabled
+        except ImportError:
+            return False
 
+        fips_enabled = is_fips_enabled()
+        if (fips_enabled):
```
Comment (alexa-phantom) [inline ewsonprem_connector.py:2600]: Don't need parenthesis

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

File: ewsonprem_connector.py (hunk 1 above)
Comment (chunmanjimmyf) [reply ewsonprem_connector.py]: Updated.

**Review by rgil-splunk — APPROVED** [review]

**Review by ricardo-phantom — APPROVED** [review]

**Comment by chunmanjimmyf** [comment]
Closed in favour of https://github.com/splunk-soar-connectors/office365/pull/13 due to merge conflicts.

**Pull request closed without merging**